### PR TITLE
fix(examples): Refactor examples for reference architectures

### PR DESCRIPTION
**Pull request by sebastianczech**
## Description

PR delivers for examples:
- `combined_design`:
- `combined_design_autoscale`
- `centralized_design`
- `centralized_design_autoscale`
- `isolated_design`
- `isolated_design_autoscale`
- `panorama_standalone`
- `vmseries_standalone`

changes:
- introduce `optional` in order to define whole type of variable `vpcs` (it requires to defined Terraform 1.3 as minimum in `versions.tf`)
- remove `vpc_subnet`, add separate keys for `vpc` and `subnet`
- simplify usage of module `subnet_sets` (without changing internal logic module)
- simplify creating VPC routes

## Motivation and Context

There were few motivations for this PR:
1. issue #37
2. internal discussion on planning meeting

As in previous year there were prepared `examples` for references architectures, there were done with below assumptions:
1. internal `modules` should not to be rebuild during preparation of `examples`
2. most of settings can be changed via modification of `example.tfvars` (e.g. new VPC, new subnet, additional route via TGW, IGW, NAT Gateway or GWLB endpoint etc.)
3. minimal version of Terraform for `modules` and `examples` remains the same (Terraform 1.0, so there is no possibility to use optionals in `examples`)
4. concepts used in `modules` (e.g. sets for subnets, NAT gateway, GWLB endpoints etc.) are not going to be changed

About the need of refactor of AWS modules we were talking multiple times, but we wanted to finish at first refactor of Azure modules - it's was finished already on April in https://github.com/PaloAltoNetworks/terraform-azurerm-swfw-modules/releases/tag/v3.0.0. 

**In order to define best user friendly `example.tfvars` (according to [the Amazon Method - Working Backwards](https://www.aboutamazon.com/news/workplace/an-insider-look-at-amazons-culture-and-processes)), refactor of internal modules, embedding some logic of existing modules in other modules (e.g. maybe `subnet` inside `vpc` module?) and rethink of `set` concept needs to be done (it's not in this PR) - this PR introduces fixes for the most painful fragments of the examples. Refactor of AWS modules will take time, this fixes can be introduced quicker.**

If you still see another improvements in examples (without rebuilding internal modules):
1. please put it as a comment and propose changes
2. or prepare your own PR with proposition of completely new approach for examples

## How Has This Been Tested?

Code was tested locally and via ChatOps.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sebastianczech** [comment]
/idempotence paths="examples/combined_design examples/combined_design_autoscale examples/isolated_design examples/isolated_design_autoscale"

**Comment by sebastianczech** [comment]
/idempotence paths="examples/centralized_design examples/centralized_design_autoscale examples/panorama_standalone examples/vmseries_standalone"

File: examples/centralized_design/main.tf [hunk 1]
```diff
@@ -17,62 +17,120 @@ module "vpc" {
...
+    associate_route_table   = [for v in value : try(v.associate_route_table, true)]               # List of associate_route_table flags
+    route_table_name        = [for v in value : try(v.route_table_name, null)]                    # List of route_table_name values
+    local_tags              = [for v in value : try(v.local_tags, {})]                            # List of local_tags maps
+  } }
+}
+
 module "subnet_sets" {
-  for_each = toset(flatten([for _, v in { for vk, vv in var.vpcs : vk => distinct([for sk, sv in vv.subnets : "${vk}-${sv.set}"]) } : v]))
+  for_each = local.subnets
   source   = "../../modules/subnet_set"
```
Comment (acelebanski) [inline examples/centralized_design/main.tf:56]: Can we perhaps move the `source` statements above `for_each` statements for each module call? Like in Azure, looks better imo.

**Review by acelebanski — APPROVED** [review]
Looks good to me! 👍

File: examples/centralized_design/main.tf (hunk 1 above)
Comment (sebastianczech) [reply examples/centralized_design/main.tf:56]: Done

**Review by lstadnik — APPROVED** [review]
LGTM

**Pull request merged**